lib/bech32/test/Codec/Binary/Bech32Spec.hs

```diff
@@ -163,9 +164,19 @@ spec = do
                     (T.length corruptedString === T.length originalString - 1)
                     .&&.
                     (Bech32.decode corruptedString `shouldSatisfy` isLeft)
+                    .||.
+                    -- In the case where the tail of a valid Bech32 string is
+                    -- composed of one or more consecutive 'q' characters
+                    -- followed by a single 'p' character, omitting any or all
+                    -- of the 'q' characters will still result in a valid
+                    -- Bech32 string:
+                    (T.length suffix > 0
+                        && T.last suffix == 'p'
+                        && char == 'q'
+                        && T.all (== 'q') (T.dropEnd 1 suffix))
```

Comment:
What's the rationale here :thinking: ?

Reply:
Good question. I'm currently updating the issue with a longer explanation. Will let you know when I've finished writing. :)

Reply:
In our current test case, we assume that deleting a character from a valid Bech32 string will create an invalid Bech32 string.

I think this is a fair assumption, as Bech32 was designed with the specific goal of **collision resistance**. When entering one Bech32 string, it's supposed to be **very hard** to collide with another Bech32 string just by making a small typo.

Experimental testing shows that there is at least one known counterexample to this law: when a Bech32 string ends in `qp`, it's possible to delete a single `q` character and still produce a valid Bech32 string. This works for any number of `q` characters.

The fact that it's possible to produce a collision just by deleting a single character comes as a surprise.

The above change adjusts the test case to account for this counterexample.